Applications/src/detect-cardiac-phases.cc

```diff
@@ -0,0 +1,132 @@
+
+char *cine_name = NULL;
+char *out_ED_name = NULL;
+char *out_ES_name = NULL;
+
+//void usage()
+//{
+//	cerr << "Usage: detect-cardiac-phases [cine] [ED output] [ES output]" << endl;
+//	exit(1);
+//}
+
+void PrintHelp(const char *name)
+{
+  cout << endl;
+  cout << "Usage: " << name << " <cine> <ED output> <ES output>" << endl;
+  cout << endl;
+  cout << "Description:" << endl;
+  cout << "  This program detects ED and ES phases from a cine image sets." << endl;
+  cout << endl;
+  cout << "Arguments:" << endl;
+  cout << "  cine        Input cine image sets." << endl;
+  cout << "  ED output   ED phase image." << endl;
+  cout << "  ES output   ES phase image." << endl;
+  cout << endl;
+  PrintCommonOptions(cout);
+  cout << endl;
+}
+
+int main(int argc, char* argv[] )
+{
+	int ok,i,j,k,t,esphase,frames;
+    short cine_max,cine_min,cinedis;
+	double *similarity,*smoothsimilarity,dif;
+	// Check command line
+	REQUIRES_POSARGS(3);
+	InitializeIOLibrary();
+
+	// Parse source and target images
+	cine_name = POSARG(1);
+	out_ED_name = POSARG(2);
+    out_ES_name = POSARG(3);
+
+    // Create images
+    GreyImage cine;
+    cine.Read(cine_name);
+    GreyImage blured;
+    blured.Initialize(cine.GetImageAttributes());
+
+    GaussianBlurring<GreyPixel> gaussianBlurring(2);
+    gaussianBlurring.Input (&cine);
+    gaussianBlurring.Output(&blured);
+    gaussianBlurring.Run();
+
+    ImageAttributes atr = cine.GetImageAttributes();
+    similarity = new double[atr._t];
+    smoothsimilarity = new double[atr._t];
+    frames = atr._t;
+    atr._t = 1;
+
+    GreyImage out_ED(atr);
+    GreyImage out_ES(atr);
+    out_ED = cine.GetFrame(0);
+    // Create similarity
+    blured.GetMinMax(&cine_min,&cine_max);
+    cinedis = cine_max - cine_min;
+    for(i = 0; i < frames; i++){
+        similarity[i] = 0;
+        smoothsimilarity[i] = 0;
+    }
+    // Evaluate similarity
+    for (t = 0; t < frames; t++){
+        for (k = 0; k < cine.GetZ(); k++){
+            for (j = 0; j< cine.GetY(); j++){
+                for (i = 0; i<cine.GetX(); i++){
+                   dif = (blured.GetAsDouble(i,j,k,t) - blured.GetAsDouble(i,j,k,0))/cinedis;
+                   similarity[t] += dif*dif;
+                }
+            }
+        }
+        cout << "similarity : " << similarity[t] << endl;
+    }
+    for(i = 0; i < frames; i++)
+        similarity[i] = sqrt(similarity[i]);
+    // Smooth similarity
+    for(i = 1; i < frames - 1; i++)
+        smoothsimilarity[i] = (similarity[i-1] + similarity[i] + similarity[i+1])/3;
+    // Find min similarity
+    dif = 0;
+    for(i = 0; i < frames; i++){
+        if(dif < smoothsimilarity[i]){
+            dif = smoothsimilarity[i];
+            esphase = i;
+        }
+    }
+
+    cout << "ES phase is: " << esphase << endl;
+
+    out_ES = cine.GetFrame(esphase);
+    // Output
+    out_ED.Write(out_ED_name);
+    out_ES.Write(out_ES_name);
+    delete []smoothsimilarity;
+    delete []similarity;
+}
```

Comment:
Could you add a newline at the EOF? That's common when editing files on Windows.